web/gridt/resources/user.py

```diff
@@ -62,8 +65,8 @@ def post(self):
             return ({"message": "Successfully changed password."}, 200)
 
 
-class ChangeEmailResource(Resource):
-    schema = ChangeEmailSchema()
+class RequestEmailChangeResource(Resource):
+    schema = RequestEmailChangeSchema()
 
     @jwt_required()
     def post(self):
```

Comment:
This function does not end in a return statement, what happens when all "if"s evaluate to false? Also what logic can be moved to schemas?

Reply:
Moving the last section out of the if statement should solve it, since we already checked whether the password was correct.

My guess is that password verification can be moved to schemas. However, checking whether an e-mail is already taken is a problem, since we might not want to throw a 400 code at this (which would be the result of putting the functionality into a schema, right?)

Reply:
Although then we need to use current_identity in schemas. Checking whether this works right now.

Reply:
As before, this doesn't work, sadly.

Reply:
Can you give me a little more context on why it does not work?

Reply:
With Schema contexts this is now solved.